spec/enum_methods_spec.rb

```diff
@@ -0,0 +1,152 @@
+# rubocop:disable Layout/LineLength
+
+require './enum_methods'
+
+RSpec.describe Enumerable do
+  let(:arr_1) { [1, 2, 3, 4, 5] }
+  let(:arr_2) { [1, 2, nil, 4] }
+  let(:arr_3) { [1, 2, 3, 4, 'cool'] }
+  let(:arr_string1) { %w[cat dog wombat] }
+
+  describe '#my_each' do
+    context 'return array if block given' do
+      it { expect(arr_1.my_each { |a| a }).to eq(arr_1) }
+    end
+    context 'if no block given return enumerator' do
+      it { expect(arr_1.my_each).to be_a(Enumerator) }
+    end
+  end
+
+  describe '#my_each_with_index' do
+    context 'if no block given return enumerator' do
+      it { expect(arr_1.my_each_with_index).to be_a(Enumerator) }
+    end
+    context 'return object after application of block given requirements' do
+      hash = {}
```

Comment:
[OPTIONAL]
- [ ] Define the hash variable on line #25 also like a let variable within the '#my_each_with_index' block